brainiak/utils/fmrisim.py

```diff
@@ -1362,15 +1461,9 @@ def generate_noise_volume(dimensions,
         # If this is below zero then all the noise will be temporal
```

Comment:
Not sure what you mean by this. I am pretty sure that temporal_sd only varies the temporal component of the noise but is limited when spatial_sd is greater

Reply:
This is a complicated part of the simulation that I could be doing wrong. The goal is to make a baseline volume of noise where each voxel has a specific mean and standard deviation. Importantly this noise will contribute to any measurement of SFNR or SNR I want to do so I need to be careful about how these different types of noise sum together. So for instance when generating this volume I need to add the spatial noise (gaussian) to the temporal noise (which is gaussian around the spatial noise). But after that addition I don't want to have changed the amount mean or variability of the spatial and temporal noise. I think that this code achieves it but I could be wrong and I would appreciate any suggestions to either fix it or make it more clear.

Reply:
Unfortunately I still did not understand the rationale of the if/else clause. Does it work without this?

Reply:
Yeah good question. The problem is that I want to subtract the spatial variance from the temporal variance but if the spatial variance is smaller than the temporal variance this would be negative. In such cases I just set the spatial variance to zero